tracer-core/src/main/java/org/zalando/tracer/FlowIDGenerator.java

```diff
@@ -15,15 +16,15 @@
 
     @Override
     public String generate() {
-        final UUID uuid = UUID.randomUUID();
+        final Random random = ThreadLocalRandom.current();
 
-        final ByteBuffer buffer = ByteBuffer.wrap(new byte[16]);
-        buffer.putLong(uuid.getMostSignificantBits());
-        buffer.putLong(uuid.getLeastSignificantBits());
+        final ByteBuffer buffer = ByteBuffer.allocate(Long.BYTES * 2);
+        buffer.putLong(random.nextLong());
+        buffer.putLong(random.nextLong());
 
         final byte[] bytes = buffer.array();
 
-        return "R" + encoder.encodeToString(bytes).replaceAll("=", "").substring(1);
+        return "R" + encoder.encodeToString(bytes).replace("=", "").substring(1);
```

Comment:
That was an interesting copy&paste issue. The legacy, internal implementation used `replaceAll` (regex) instead of `replace` (full match string replace), for some reason.

Reply:
Though replace is implemented with Pattern anyway :)

Reply:
It is?

Reply:
I guess you actually need this here https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Encoder.html#withoutPadding--

Reply:
You can also encode to `byte[]`, set first byte to `R` and then construct `new String(bytes)`